### PR TITLE
Fixed bug #24644: Units moving after in game help exit 

### PR DESCRIPTION
**Pull request by qmo2015**
Fixed bug #24644: Units moving after in game help exit 
https://gna.org/bugs/index.php?24644

### CODE REVIEW

File: src/mouse_events.cpp [hunk 1]
```diff
@@ -470,9 +470,10 @@ bool mouse_handler::right_click_show_menu(int x, int y, const bool /*browse*/)
 
 void mouse_handler::select_or_action(bool browse)
 {
-	if (!pc_.get_map_const().on_board(last_hex_))
+	if (!pc_.get_map_const().on_board(last_hex_)){
 		tooltips::click(drag_from_x_, drag_from_y_); 
 		return;
+		}
```
Comment (CelticMinstrel) [inline src/mouse_events.cpp]: This should be indented only one level, not two.

Comment (qmo2015) [inline src/mouse_events.cpp]: The indentation is now fixed.

**Comment by qmo2015** [comment]
The detail of my fix: 
I noticed that left_click() function in mouse_handler_base.cpp and select_or_action() function in mouse_events.cpp would be called after a click. left_click() checks if the click was on the mini map or an ability name on the sidebar and select_or_action() selects a spot on the map where the current location of the cursor is. What was causing the bug was that the function call tooltips::click(x, y) in left_click opens the help menu and changes the location of the cursor, because players had to move the cursor to the "close" button to close the menu, which leads to select_or_action() selects an unexpected spot on the map. I moved the function call tooltips::click(x, y) from left_click to select_or_action and made select_or_action return after the function call. That way select_or_action would not select a spot after the help menu was closed.

**Comment by CelticMinstrel** [comment]
By the way, just for future reference, it's typically better to open pull requests from a branch other than master, so that other random stuff you push to master doesn't end up in the pull request. This doesn't matter to us, it's just a general tip.

(However, I do suggest you don't push anything else until this is merged, to avoid potentially messing it up.)

I'm also happy to help if you're not sure how to create a branch.

**Comment by CelticMinstrel** [comment]
(Oh, as one exception, you can push a commit adding yourself under Miscellaneous Contributors in the credits, about.cfg. Also a changelog entry if you feel it's important.)

**Comment by qmo2015** [comment]
Ok, I added my name to credits and entry to changelog.

**Comment by CelticMinstrel** [comment]
We prefer to have the issue number in parentheses at the end, like the example two lines down.

**Comment by qmo2015** [comment]
I put the bug number in parentheses at the end.

**Comment by Wedge009** [comment]
This works for me, confirm bug #24644 is resolved with this.

**Pull request merged**